### DISCUSSION TITLE
Can this library be used in React Native?

### DISCUSSION THREAD

**Discussion by nditanaka** [reactivemarkets/react-financial-charts#463] [Q&A] [answered]
I am working on a financial charts app in React Native and I'm currently using [react-google-charts](https://react-google-charts.com/) to build my candlestick (OHLC) charts in a react native Webview but the charts look unsightly on mobile. I'm wondering if this might be a better solution to build chart components natively and not through a Webview?

Thank you!

**Comment by markmcdowell** [comment, marked as answer]
I'm interested in investigating it, the problem I see is the library relies on canvas which would need the Webview.

**Reply by nditanaka** [reply]
I see! I am trying to avoid using WebView because I am having responsiveness issues with my current WebView: https://snack.expo.io/@nditanaka/react-google-charts-webview

As you can see, the charts are distorted on mobile but they display correctly on Desktop. Are you also trying to create candlestick charts? I am looking into using d3 from William Candillon's Can it be done in RN course: https://www.youtube.com/watch?v=r5XXSb4yQes&ab_channel=WilliamCandillon

**Reply by markmcdowell** [reply]
The expo didn’t work, it could be an issue with the charting lib rather than web view though?

from the video I can see he uses react-native-svg, that does sound doable for this project. Just involves how to change the library to build for both web and rn.

**Reply by nditanaka** [reply]
There was an issue with the Chart component. I switched it back to a WebView. It's working right now.

**Reply by venomkage** [reply]
Hey, I also want to implement a candlestick chart in my react-native app. Can you please help me out?
